### PR TITLE
Move Predictive handlers to observational module

### PR DESCRIPTION
**Pull request by eb8680**
Blocked by #549

This pure refactoring PR moves some of the utilities for making and batching predictive distributions out of `chirho.robust.handlers.predictive` and `chirho.robust.internals.nmc` and into a new `chirho.observational.handlers.predictive`. There are no changes to the code other than moving definitions to different files (with one small exception, noted below).

### CODE REVIEW

File: chirho/robust/internals/nmc.py
Comment (eb8680) [inline chirho/robust/internals/nmc.py:165]: This is the only non-renaming change in this PR. It is equivalent to the previous version with the definition of `bind_leftmost_dim` inlined for this special case.

Comment (SamWitty) [reply chirho/robust/internals/nmc.py:165]: I have a slight preference for keeping it as `bind_leftmost_dim`, as that seems a bit more transparent about what the code is doing.

If we do decide to remove it here is `bind_leftmost_dim` used anywhere else anymore?

**Review by SamWitty — CHANGES_REQUESTED** [review]
As noted I have a slight preference to revert the `bind_leftmost_dim` change, but I'll happily defer to your preferences here.

I marked this as "request changes" just because the linter seems to be failing because of unused imports. Once that's resolved I'll approve.

File: chirho/robust/internals/nmc.py
Comment (eb8680) [reply chirho/robust/internals/nmc.py:165]: I inlined `bind_leftmost_dim` here because it is now in `observational.internals` and shouldn't be used outside `observational`. I could have moved `BatchedNMCLogMarginalLikelihood`, but I would like to keep that as an internal implementation detail of `chirho.robust` for now.

`bind_leftmost_dim` is still used inside `chirho.observational.handlers.predictive.PredictiveFunctional`, which was moved in this PR.

**Review by SamWitty — APPROVED** [review]

**Pull request merged**
